### PR TITLE
Documenting worker pool

### PR DESCRIPTION
**Pull request by pxp9**

### CODE REVIEW

File: src/asynk/async_worker_pool.rs [hunk 1]
```diff
@@ -8,19 +8,27 @@ use log::error;
...
     AQueue: AsyncQueueable + Clone + Sync + 'static,
 {
     #[builder(setter(into))]
+    /// AsyncWorkerPool uses a queue to control the tasks that will be executing.
     pub queue: AQueue,
+    /// sleep_params controls how much time tokio process sleeps when they are not tasks to
+    /// execute.
     #[builder(default, setter(into))]
     pub sleep_params: SleepParams,
+    /// retention_mode controls the treatment of the tasks once they are finished.
```
Comment (ayrat555) [inline src/asynk/async_worker_pool.rs:25]:
```suggestion
    /// retention_mode controls if  tasks should be persisted after execution
```

File: src/asynk/async_worker_pool.rs [hunk 2]
```diff
@@ -8,19 +8,27 @@ use log::error;
 use tokio::task::JoinHandle;
 use typed_builder::TypedBuilder;
 
+/// A executioner of tasks, it executes tasks only of one given task_type, it sleeps when they are
```
Comment (ayrat555) [inline src/asynk/async_worker_pool.rs:11]: I think this doc is related to worker, not worker pool

Comment (pxp9) [reply src/asynk/async_worker_pool.rs]: [move comment to worker](https://github.com/ayrat555/fang/pull/104/commits/44e833b300f6f60ddc946ec23c0e87bc34ecd6ec)

File: src/asynk/async_worker_pool.rs [hunk 3]
```diff
@@ -8,19 +8,27 @@ use log::error;
...
+/// not tasks to be executed.
 #[derive(TypedBuilder, Clone)]
 pub struct AsyncWorkerPool<AQueue>
```
[the remark this answers is not in the document]
Comment (pxp9) [reply src/asynk/async_worker_pool.rs]: [documenting blocking module](https://github.com/ayrat555/fang/pull/104/commits/44359e0fc9510118cf78c2af2bfe4257ba217590)


maybe it is not all blocking module documented, review the docs please.

File: src/asynk/async_worker_pool.rs [hunk 4]
```diff
@@ -14,13 +14,19 @@ where
     AQueue: AsyncQueueable + Clone + Sync + 'static,
 {
     #[builder(setter(into))]
+    /// the AsyncWorkerPool uses a queue to control the tasks that will be executed.
     pub queue: AQueue,
+    /// sleep_params controls how much time a worker will sleep while waiting for tasks
+    /// execute.
```
Comment (ayrat555) [inline src/asynk/async_worker_pool.rs:20]:
```suggestion
```

File: src/blocking/queue.rs [hunk 5]
```diff
@@ -84,29 +84,46 @@ impl From<cron::error::Error> for QueueError {
     }
 }
 
+/// This trait defines operations for an synchronous queue.
```
Comment (ayrat555) [inline src/blocking/queue.rs:87]:
```suggestion
/// This trait defines operations for a synchronous queue.
```

File: src/asynk/async_worker.rs [hunk 6]
```diff
@@ -9,6 +9,8 @@ use crate::{RetentionMode, SleepParams};
 use log::error;
 use typed_builder::TypedBuilder;
 
+/// A executioner of tasks, it executes tasks only of one given task_type, it sleeps when they are
+/// not tasks to be executed.
```
Comment (ayrat555) [inline src/asynk/async_worker.rs:13]:
```suggestion
```

File: src/blocking/queue.rs [hunk 7]
```diff
@@ -117,6 +134,27 @@ pub trait Queueable {
     ) -> Result<Task, QueueError>;
 }
 
+/// An async queue that can be used to enqueue tasks.
+/// It uses a PostgreSQL storage. It must be connected to perform any operation.
+/// To connect an `Queue` to PostgreSQL database call the `get_connection` method.
```
Comment (ayrat555) [inline src/blocking/queue.rs:139]:
```suggestion
/// To connect a `Queue` to the PostgreSQL database call the `get_connection` method.
```

**Review by ayrat555 — APPROVED** [review]

**Pull request merged**